### DISCUSSION TITLE
Is it possible to change the production branch using Vercel API or CLI?

### DISCUSSION THREAD

**Discussion by redshoga** [vercel/vercel#5869] [Help] [answered]
As the title says, I would like to change the production branch by API for the operation of the project.

Reference
https://vercel.com/docs/git#production-branch

**Comment by styfle** [comment, marked as answer]
We don't have the API documented currently, however you can use the Repository Default branch and then use your git provider's API (for example GitHub) to change the default branch.

<img src="https://user-images.githubusercontent.com/229881/108532470-e14ca900-72a5-11eb-8149-20164f21a8ed.png" width=300 />

Is this something you plan to change regularly?

**Reply by redshoga** [reply]
Thank you for your answer. :)

I sometimes use the production branch setting so that a project can be tested on a particular domain without merging branches.

It would be very helpful if it were implemented as an API.

**Reply by styfle** [reply]
You can achieve this without changing your production domain.

Instead, you can assign a git branch to a preview domain.

https://vercel.com/docs/custom-domains#assigning-a-domain-to-a-git-branch
